linux/just_files/just_pre-commit_functions.bsh

```diff
@@ -0,0 +1,294 @@
+  local project_cwd="${JUST_PROJECT_PREFIX}_CWD"
+  project_cwd=${!project_cwd}
+  local precommit_dir="${JUST_PROJECT_PREFIX}_PRECOMMIT_DIR"
+  precommit_dir=${!precommit_dir:-${project_cwd}/.pre-commit}
+  local python_dir="${JUST_PROJECT_PREFIX}_PRECOMMIT_PYTHON_DIR"
+  python_dir=${!python_dir-${precommit_dir}/python}
+  local desired_python_version="${JUST_PROJECT_PREFIX}_PRECOMMIT_PYTHON_VERSION"
+  desired_python_version=${!desired_python_version-3.12.1}
+
+  local hook_types="${JUST_PROJECT_PREFIX}_PRECOMMIT_HOOKS[@]"
+  # This won't recognize an array set as empty as a set state, it will be treated as unset
+  hook_types=("${!hook_types-pre-push}")
+
+  arg="${1}"
+  shift 1
+  case ${arg} in
+    pre-commit) # Run pre-commit command (e.g. just pre-commit run)
+      pre-commit "${@}"
+      extra_args=${#}
+      ;;
+    pre-commit_run) # Run pre-commit check locally. Defaults to "-a" to run everything
+      if (( ${#} )); then
+        pre-commit run "${@}"
+        extra_args=${#}
+      else
+        pre-commit run -a
+      fi
+      ;;
+    pre-commit_init) # Initialize pre-commit for the first time in a new project
+      mkdir -p "${precommit_dir}"
+      cp "${VSI_COMMON_DIR}/templates/pre-commit/requirements.in" "${precommit_dir}/requirements.in"
+      cp "${VSI_COMMON_DIR}/templates/pre-commit/config.yml.in" "${precommit_dir}/config.yml.in"
+
+      local precommit_rel=$(relative_path "${precommit_dir}" "${project_cwd}")
+      local precommit_rel_regex=$(regex_escape "${precommit_rel}")
+      if ! grep '^'"${precommit_rel_regex}"'/config.yml$' .gitignore &> /dev/null; then
+        sed -i '$a'"${precommit_rel}"'/config.yml' .gitignore
+      fi
+      if ! grep '^'"${precommit_rel_regex}"'/python$' .gitignore &> /dev/null; then
+        sed -i '$a'"${precommit_rel}"'/python' .gitignore
+      fi
+
+      echo "Don't forget to run: git add .pre-commit .gitignore" >&2
+      ;;
+    pre-commit_setup) # Setup the pre-commit python
+      local python_exe
+
+      # If pip is missing, install python
+      if [ ! -f "$(vsi::python::virtualenv_bin "${python_dir}" pip)" ]; then
+        local PYTHON_VERSION=${PYTHON_VERSION-${desired_python_version}}
+        # conda-python-install sets: python_exe conda_exe python_activate python_version
+        local python_exe conda_exe python_activate python_version
+        conda-python-install --dir "${python_dir}" --download
+      fi
+
+      # If pip-sync is missing, install just pip-tools
+      if [ ! -f "$(vsi::python::virtualenv_bin "${python_dir}" pip-sync)" ]; then
+        local requirements="${precommit_dir}/requirements.txt"
+        if [ ! -f "${requirements}" ]; then
+          # Bootstrap case
+            requirements="${precommit_dir}/requirements.in"
+        fi
+        "$(vsi::python::virtualenv_bin "${python_dir}" pip)" \
+            install -c "${requirements}" pip-tools
+      fi
+      ;;
+    pre-commit_sync) # Sync the virtualenv pre-commit and plugins runs in
```

Comment:
what is this saying? maybe take off "runs in"

Reply:
```suggestion
    pre-commit_sync) # Sync the virtualenv for pre-commit and plugins
```